dockerTools: update cross compilation support of the pullImage function (#186975)
NixOS/nixpkgs: doc/builders/images/dockertools.section.md

```nix
pullImage {
  imageName = "docker.io/library/busybox";
  imageDigest = "sha256:ef320ff10026a50cf5f0213d35537ce0041ac1d96e9b7800bafd8bc9eff6c693";
  sha256 = "sha256-9T6tP/xnZwaTH+qkbkVuXhCk+qZGE3WBY8T+Ug4qCmc=";

>   finalImageName = "docker.io/library/busybox";

Reviewer: ```suggestion

  # optional parameters
  finalImageName = "docker.io/library/busybox";
```

Would be nice too see at a glance which ones are required and which ones aren't.

  finalImageTag = "stable-uclibc";
  os = "linux";

>   arch ="arm";

Reviewer: ```suggestion
  arch = "arm";
```

- `sha256` is the checksum of the whole fetched image. This argument is required.

> The host information (arch and cpu variant if available) will be used to fetch the correct image. The target os is set to `linux` by default and can't be changed to another os.

Reviewer: `target` has a specific meaning in the context of cross compilation, where it's only relevant when compiling a compiler.

```suggestion
The host information (arch and cpu variant if available) will be used to fetch the correct image. You can fetch images for a different platform by using the Nixpkgs cross compilation functionality.
```

I don't think we have to warn about the missing functionality if there is any.

Seems to be `GOOS` actually https://github.com/opencontainers/image-spec/blob/main/config.md#properties

Author: Thanks for the tip, I didn't know the special meaning of that word in this context :sweat_smile:
